pkg/apis/cyclone/v1alpha1/workflow_param.go

```diff
@@ -0,0 +1,25 @@
+package v1alpha1
+
+import metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+
+// +genclient
+// +genclient:noStatus
+// +k8s:deepcopy-gen:interfaces=k8s.io/apimachinery/pkg/runtime.Object
+
+// WorkflowParam describes global workflow runtime parameters
+type WorkflowParam struct {
+	// Metadata for the resource, like kind and apiversion
+	metav1.TypeMeta `json:",inline"`
+	// Metadata for the particular object, including name, namespace, labels, etc
+	metav1.ObjectMeta `json:"metadata,omitempty"`
+	// Workflow param specification
+	Spec WorkflowParamSpec `json:"spec"`
+}
+
+// WorkflowParamSpec describes global parameters a workflow should run with.
+type WorkflowParamSpec struct {
+	// PVC to be used for workflow
+	PVC string `json:"pvc"`
```

Comment:
Is this used for cache purpose? do i have to pass a PVC, what if the parameter is empty?

Reply:
Yes, this is used for transfer data (e.g. artifacts) within workflow from one stage to another. If no such need, it can be omitted. 

How about letting users decide whether to set it or not.

Reply:
didn't we already have persistent flag? sounds a bit confusing and duplicated, even though they have different purpose.

Reply:
This PVC is a tmp pvc, it's necessary for the workflow implementation. While the resource persistence is using another dedicated resource PV, resource management (including the PV.) is independent to workflow.  For a workflow, resources (input, output) are different to artifacts.

Reply:
Yeah, I understand the difference. Using PVC here means we give control to our users, but it's fundamentally a temp PVC which means user shouldn't even care about it.

IMO, the API is not consistent with what PVC should do...